library/core/class.oauth2.php

```diff
@@ -455,11 +471,9 @@ public function authorizeUri($state = []) {
      *
      * @return string Endpoint of the provider.
      */
-    protected function realAuthorizeUri(array $state = []): string {
+    protected function realAuthorizeUri(string $uri, array $state = []): string {
```

Comment:
Won't adding a new parameter in front of an existing parameter break backwards compatibility with classes that have extended this one?

Reply:
This method has only been available for a 5 months. We have not extended it in any of our plugins. Do you think it could be a risk for open source?

Reply:
That is a very real possibility, but it's usually a good habit to avoid this type of thing, in general.

Reply:
I agree with Ryan. Usually the refactor is to do something like this.

1. Make a `realAuthorizeUriInternal` method with the signature like you have here (please mark this final).
2. Make `realAuthorizeUri` a 1-liner that calls `realAuthorizeUriInternal`.
3. Make another 1-liner `realRegisterUri` that calls `realAuthorizeUriInternal`.